src/app/static/src/js/components/runWorkflow/runWorkflowRun.vue

```diff
@@ -1,73 +1,197 @@
 <template>
-    <div>
+    <div id="run-workflow-id">
         <h2 id="run-header">Run workflow</h2>
         <div id="workflow-name-div" class="form-group row">
             <label class="col-sm-4 col-form-label text-left">Name</label>
             <div class="col-sm-4">
-                <input type="text" class="form-control" id="run-workflow-name" placeholder="Impact estimates">
+                <input type="text"
+                       :readonly="isRerun"
+                       @input="handleValidation"
+                       v-model="workflowName"
+                       class="form-control"
+                       id="run-workflow-name"
+                       placeholder="Impact estimates">
+                <small class="text-danger">{{workflowNameError}}</small>
             </div>
         </div>
-        <div id="workflow-source-div" class="form-group row">
-            <label class="col-sm-4 col-form-label text-left">Database "source"</label>
-            <div class="col-sm-4">
-                <input type="text" class="form-control" id="run-workflow-database_source" placeholder="Production">
+        <template v-if="showInstances">
+            <div v-for="(options, name) in runMetadata.metadata.instances" v-if="options.length > 1"
+                 id="workflow-source-div" class="form-group row">
```

Comment:
'source' is just one possible name for an instances - would a better id be `workflow-instances-div`? Would also mean updating the tests which refer to 'workflow-source'